openauto/Projection/GSTVideoOutput.cpp

```diff
@@ -49,28 +48,13 @@ GSTVideoOutput::GSTVideoOutput(configuration::IConfiguration::Pointer configurat
 
     videoSink_ = surface_->videoSink();
 
+
     GError* error = nullptr;
-    const char* vidLaunchStr = "appsrc name=mysrc is-live=true block=false max-latency=100 do-timestamp=true stream-type=stream ! queue ! h264parse ! "
-        #ifdef RPI
-            #ifdef PI4
-                               "v4l2h264dec ! "
-            #else
-                               "omxh264dec ! "
-            #endif
-        #else
-                               "avdec_h264 ! "
-        #endif
-                                "videocrop top=0 bottom=0 name=videocropper ! capsfilter caps=video/x-raw name=mycapsfilter";
-    #ifdef RPI
-        OPENAUTO_LOG(info) << "[GSTVideoOutput] RPI Build, running with " <<
-        #ifdef PI4
-                              "v4l2h264dec";
-        #else
-                              "omxh264dec";
-        #endif
-    #endif
+    std::string vidLaunchStr = "appsrc name=mysrc is-live=true block=false max-latency=100 do-timestamp=true stream-type=stream ! queue ! h264parse ! capssetter caps=\"video/x-h264,colorimetry=bt709\" ! ";
+    vidLaunchStr += ToPipeline(findPreferredVideoDecoder());
+    vidLaunchStr += " ! videocrop top=0 bottom=0 name=videocropper ! capsfilter caps=video/x-raw name=mycapsfilter";
     
-    vidPipeline_ = gst_parse_launch(vidLaunchStr, &error);
+    vidPipeline_ = gst_parse_launch(vidLaunchStr.c_str(), &error);
```

Comment:
assuming the `GstElement *` returned by `gst_parse_launch` doesnt need `vidLaunchStr.c_str()` to live beyond this call?

since `vidLaunchStr` goes out of scope when you return from this function

Reply:
Yea the launch string doesn't need to live - it's the instruction manual used to build the pipeline, but isn't needed once the pipeline exists.